src/treelearner/feature_histogram.hpp

```diff
@@ -240,16 +276,29 @@ class FeatureHistogram {
     }
 
     if (is_splittable_) {
-      output->left_output = CalculateSplittedLeafOutput(best_sum_left_gradient, best_sum_left_hessian,
-                                                        meta_->config->lambda_l1, l2, meta_->config->max_delta_step,
-                                                        min_constraint, max_constraint);
+      SplittingConstraint *left;
+      SplittingConstraint *right;
+      if (constraints != nullptr) {
+        left = &(constraints->left);
+        right = &(constraints-> right);
+      }
+      else {
+        left =  new SplittingConstraint();
+        right =  new SplittingConstraint();
```

Comment:
maybe initialize `left` and `right` to nullptr? and handle the nullptr in `CalculateSplittedLeafOutput` function?